File: 04/factorial.rb
################################
# Name: Lily McAboy
# Date: 2/14/2024
# CS 214 Project 04
################################

if __FILE__ == $0
  print "Enter your desired factorial number: "
  user_num = gets.chomp.to_f

  fact = 1
  for i in 1.. user_num do
    fact = fact * i
  end

  print "The factorial of "
  print user_num 
  print " is "
  print fact
  print "\n"

end

File: 04/clojure/src/factorial.clj
;;;; Name: Lily McAboy
;;;; Date: 2/14/2024
;;;; CS 214 Project 04
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns factorial)                                 ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn factorial[usr_num]
  (if (<= usr_num 1)
    1 ; base case
    (* usr_num (factorial(- usr_num 1))) ;multiply the number by the recursive function of the number -1 and etc. 
    )
  )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn -main []
  (println "Computing the factorial of a user given number.\n")
  (print "Enter your desired factorial number: ") (flush)
  (let
    [ n (int(read))]
    (printf "The factorial of %d is %d \n" n (factorial n))
    )
  )

              
